tests/e2e/test_api.py

```diff
@@ -804,42 +805,53 @@ def test_feedback_storing_standalone():
-#     # for this test
-#     feedbacks = cluster_utils.list_path(pod_name, feedbacks_path)
-#     if feedbacks:
-#         cluster_utils.remove_dir(pod_name, feedbacks_path)
-#         assert cluster_utils.list_path(pod_name, feedbacks_path) == []
-
-#     response = client.post(
-#         "/v1/feedback",
-#         json={
-#             "conversation_id": CONVERSATION_ID,
-#             "user_question": "what is OCP4?",
-#             "llm_response": "Openshift 4 is ...",
-#             "sentiment": 1,
-#         },
-#         timeout=BASIC_ENDPOINTS_TIMEOUT,
-#     )
-
-#     assert response.status_code == requests.codes.ok
-
-#     feedback_data = cluster_utils.get_single_existing_feedback(pod_name, feedbacks_path)
-
-#     assert feedback_data["user_id"]  # we don't care about actual value
-#     assert feedback_data["conversation_id"] == CONVERSATION_ID
-#     assert feedback_data["user_question"] == "what is OCP4?"
-#     assert feedback_data["llm_response"] == "Openshift 4 is ..."
-#     assert feedback_data["sentiment"] == 1
+@pytest.mark.cluster()
+def test_feedback_storing_cluster():
+    """Test if the feedbacks are stored properly."""
+    try:
+        feedbacks_path = OLS_USER_DATA_PATH + "/feedback"
+        pod_name = cluster_utils.get_single_existing_pod_name()
+
+        # disable collector script to avoid interference with the test
+        cluster_utils.create_file(pod_name, OLS_COLLECTOR_DISABLING_FILE, "")
+
+        # there are multiple tests running agains cluster, so transcripts
+        # can be already present - we need to ensure the storage is empty
+        # for this test
+        feedbacks = cluster_utils.list_path(pod_name, feedbacks_path)
+        if feedbacks:
+            cluster_utils.remove_dir(pod_name, feedbacks_path)
+            assert cluster_utils.list_path(pod_name, feedbacks_path) == []
+
+        response = client.post(
+            "/v1/feedback",
+            json={
+                "conversation_id": CONVERSATION_ID,
+                "user_question": "what is OCP4?",
+                "llm_response": "Openshift 4 is ...",
+                "sentiment": 1,
+            },
+            timeout=BASIC_ENDPOINTS_TIMEOUT,
+        )
+
+        assert response.status_code == requests.codes.ok
+
+        feedback_data = cluster_utils.get_single_existing_feedback(
+            pod_name, feedbacks_path
+        )
+
+        assert feedback_data["user_id"]  # we don't care about actual value
+        assert feedback_data["conversation_id"] == CONVERSATION_ID
+        assert feedback_data["user_question"] == "what is OCP4?"
+        assert feedback_data["llm_response"] == "Openshift 4 is ..."
+        assert feedback_data["sentiment"] == 1
+
+    finally:
+        # ensure script is enabled again after test (succesfull or not)
+        cluster_utils.remove_file(pod_name, OLS_COLLECTOR_DISABLING_FILE)
+        assert "disable_collector" not in cluster_utils.list_path(
```

Comment:
@tisnik do you know if having a passing assert execute after a failing assert, will result in the test being reported as a pass?

i'm wondering if one of the asserts above fails, so we jump into the finally block, and then this assert passes, that we'll report the overall test as a pass.  I would hope not, but i feel like the question has to be asked :)

Reply:
I assume the original exception that resulted from the failed assert still ends up getting thrown and thus pytest records the failure